nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb

```diff
@@ -151,21 +152,35 @@ def add_transitive_dependencies_from_project_references(project_file, doc, depen
             full_path = File.expand_path(relative_path)
             full_path = full_path[1..-1] unless is_rooted
 
-            referenced_file = dependency_files.find { |f| f.name == full_path }
-            next unless referenced_file
-
-            dependency_set(project_file: referenced_file).dependencies.each do |dep|
-              dependency = Dependency.new(
-                name: dep.name,
-                version: dep.version,
-                package_manager: dep.package_manager,
-                requirements: []
-              )
-              dependency_set << dependency
+            # add full path to an array of paths
+            full_paths = handle_wildcard_path_in_project_references(full_path, dependency_set)
+
+            full_paths.each do |path|
+              referenced_file = dependency_files.find { |f| f.name == path }
+              next unless referenced_file
+
+              dependency_set(project_file: referenced_file).dependencies.each do |dep|
+                dependency = Dependency.new(
+                  name: dep.name,
+                  version: dep.version,
+                  package_manager: dep.package_manager,
+                  requirements: []
+                )
+                dependency_set << dependency
+              end
             end
           end
         end
 
+        def handle_wildcard_path_in_project_references(full_path, _dependency_set)
+          return [full_path] unless full_path.include?("*")
```

Comment:
Technically [`?` is also a wild-card](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-items?view=vs-2022#use-wildcards-to-specify-items) (though I don't believe it's common).